### ISSUE TITLE
Obscure font rendering issue when rendering with DirectX on HiDPI screen

### ISSUE THREAD

**Issue by ALANVF** [vim/vim#8807]
**Describe the bug**
When using DirectX rendering in GVIm on an HiDPI screen (in my case, a 4K laptop screen), the font weight is continually increased until Ctrl+L is pressed. Typing anything else will actually increase the font weight faster with every key pressed.
(Issues related to the runtime files should be reported to their maintainer, check the file header.)

**To Reproduce**
Detailed steps to reproduce the behavior:
1. Add `set rop=type:directx` to your .vimrc
2. Open GVim in any way
3. Font weight will increase forever, or until you close the program

**Expected behavior**
Render properly

**Screenshots**
![image](https://user-images.githubusercontent.com/16247881/131181878-59c5dc4b-8db0-4415-8d42-442eefb79976.png)

**Environment (please complete the following information):**
 - Vim version: 8.2.0
 - OS: Windows 10 20H2 build 19042
 - Terminal: GUI

**Additional context**
Seems to be related to https://github.com/vim/vim/issues/1059

**Comment by chrisbra** [comment]
just to make clear, this also happens when started with `gvim --clean`?

**Comment by ALANVF** [comment]
Yes, once I enable the renderoptions setting it happens instantly.
